src/cls/rgw_gc/cls_rgw_gc.cc

```diff
@@ -276,60 +310,50 @@ static int cls_rgw_gc_queue_remove_entries(cls_method_context_t hctx, bufferlist
     unsigned int index = 0;
     // If data is not empty
     if (op_ret.entries.size()) {
-      for (auto it : op_ret.entries) {
+      for (const auto& it : op_ret.entries) {
```

Comment:
for remove operation, we fetch one extra entry to calculate the marker up to which the entries are to be deleted, so during a requeue op, we should omit the last entry, since in the next remove, the corresponding urgent data would have been removed, hence this entry will not be skipped and will be deleted, hence will not actually be deferred - this will happen very rarely though.

Reply:
i'm not sure this can happen, since we're batching up the entries-to-requeue in `requeue_buffers`, and only writing them back to the queue after the call to `queue_remove_entries()` with `end_marker`. this `queue_list_entries()` loop won't ever see the requeued entries

Reply:
The last+1 entry wont be removed from the queue, but if it has an urgent data entry, it will get removed based on the current logic . Then when a new remove call comes in, processing will start from last+1 entry, and if it was supposed to be skipped, it wont be.

Reply:
ok, i see what you mean with the `list_op.max = op.num_entries + 1; // +1 to get the offset of last + 1 entry` thing